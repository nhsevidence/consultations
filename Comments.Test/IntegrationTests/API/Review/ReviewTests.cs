using System;
using System.Collections.Generic;
using System.Linq;
using System.Net;
using System.Net.Http;
using System.Text;
using System.Threading.Tasks;
using Comments.Configuration;
using Comments.Models;
using Comments.Services;
using Comments.Test.Infrastructure;
using Comments.ViewModels;
using Newtonsoft.Json;
using Shouldly;
using Xunit;

namespace Comments.Test.IntegrationTests.API.Review
{
    public class ReviewTests : TestBase
    {
	    private static ReviewConfig GetTestReviewConfig()
	    {
		    return new ReviewConfig() { Filters = new List<FilterGroup>()
		    {
				new FilterGroup(){ Id = "Type", Title = "Response type", Options = new List<FilterOption>()
				{
					new FilterOption("Questions", "Questions"),
					new FilterOption("Comments", "Comments"),
				}},
				new FilterGroup(){Id =  "Document", Title = "Document", Options = new List<FilterOption>()}
		    } };
		}

        [Fact]
        public async Task Get_AllCommentsForConsultationForReview()
        {
            // Arrange
            ResetDatabase();
	        _context.Database.EnsureCreated();
	        AppSettings.ReviewConfig = GetTestReviewConfig();
			var sourceURI = "consultations://./consultation/1/document/1/chapter/introduction";
	        var relativeURIForReviewPage = "/1/review";
			var answerText = Guid.Empty.ToString();
	        var questionText = Guid.Empty.ToString();
	        var userId = Guid.Empty;

	        AddCommentsAndQuestionsAndAnswers(sourceURI, "My Comment", questionText, answerText, userId);
	        AddCommentsAndQuestionsAndAnswers(sourceURI, "My Second Comment", questionText, answerText, userId);
	        AddCommentsAndQuestionsAndAnswers(sourceURI, "Another users Comment", questionText, answerText, Guid.NewGuid());

			// Act
			var response = await _client.GetAsync(string.Format("/consultations/api/CommentsForReview?relativeURL={0}", relativeURIForReviewPage));
            response.EnsureSuccessStatusCode();
            var responseString = await response.Content.ReadAsStringAsync();
            var deserialisedObject = JsonConvert.DeserializeObject<ViewModels.ReviewPageViewModel>(responseString);

			// Assert
			response.StatusCode.ShouldBe(HttpStatusCode.OK);
	        responseString.ShouldMatchApproved(new Func<string, string>[] { Scrubbers.ScrubLastModifiedDate, Scrubbers.ScrubIds });
			deserialisedObject.CommentsAndQuestions.Comments.Count().ShouldBe(2);
		}
	}
}
