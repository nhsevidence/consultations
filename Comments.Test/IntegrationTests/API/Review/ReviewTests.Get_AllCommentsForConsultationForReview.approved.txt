{
    "commentsAndQuestions":{
        "comments":[
            {
                "commentId":"scrubbed by ScrubCommentId",
                "lastModifiedDate":"scrubbed by ScrubLastModifiedDate",
                "lastModifiedByUserId":"00000000-0000-0000-0000-000000000000",
                "commentText":"My Second Comment",
                "status":{
                    "statusId":1,
                    "name":"Draft"
                },
                "statusId":1,
                "commentId":"scrubbed by ScrubLocationId",
                "sourceURI":"consultations://./consultation/1/document/1/chapter/introduction",
                "htmlElementID":null,
                "rangeStart":null,
                "rangeStartOffset":null,
                "rangeEnd":null,
                "rangeEndOffset":null,
                "quote":null,
                "commentOn":"Chapter",
                "commentOnEnum":2,
                "documentId":1,
                "show":true,
                "order":"0",
                "section":null
            },
            {
                "commentId":"scrubbed by ScrubCommentId",
                "lastModifiedDate":"scrubbed by ScrubLastModifiedDate",
                "lastModifiedByUserId":"00000000-0000-0000-0000-000000000000",
                "commentText":"My Comment",
                "status":{
                    "statusId":1,
                    "name":"Draft"
                },
                "statusId":1,
                "commentId":"scrubbed by ScrubLocationId",
                "sourceURI":"consultations://./consultation/1/document/1/chapter/introduction",
                "htmlElementID":null,
                "rangeStart":null,
                "rangeStartOffset":null,
                "rangeEnd":null,
                "rangeEndOffset":null,
                "quote":null,
                "commentOn":"Chapter",
                "commentOnEnum":2,
                "documentId":1,
                "show":true,
                "order":"0",
                "section":null
            }
        ],
        "questions":[
            {
                "questionId":"scrubbed by ScrubQuestionId",
                "questionText":"00000000-0000-0000-0000-000000000000",
                "questionTypeId":"scrubbed by ScrubQuestionTypeId",
                "lastModifiedByUserId":"00000000-0000-0000-0000-000000000000",
                "lastModifiedDate":"scrubbed by ScrubLastModifiedDate",
                "questionType":{
                    "questionTypeId":"scrubbed by ScrubQuestionTypeId",
                    "description":"text",
                    "hasTextAnswer":true,
                    "hasBooleanAnswer":false
                },
                "answers":[
                    {
                        "answerId":"scrubbed by ScrubAnswerId",
                        "answerText":"00000000-0000-0000-0000-000000000000",
                        "answerBoolean":null,
                        "questionId":"scrubbed by ScrubQuestionId",
                        "lastModifiedDate":"scrubbed by ScrubLastModifiedDate",
                        "lastModifiedByUserId":"00000000-0000-0000-0000-000000000000",
                        "statusId":1,
                        "status":{
                            "statusId":1,
                            "name":"Draft"
                        },
                        "sourceURI":"consultations://./consultation/1/document/1/chapter/introduction"
                    }
                ],
                "commentId":"scrubbed by ScrubLocationId",
                "sourceURI":"consultations://./consultation/1/document/1/chapter/introduction",
                "htmlElementID":null,
                "rangeStart":null,
                "rangeStartOffset":null,
                "rangeEnd":null,
                "rangeEndOffset":null,
                "quote":null,
                "commentOn":"Chapter",
                "commentOnEnum":2,
                "documentId":1,
                "show":true,
                "order":"0",
                "section":null
            },
            {
                "questionId":"scrubbed by ScrubQuestionId",
                "questionText":"00000000-0000-0000-0000-000000000000",
                "questionTypeId":"scrubbed by ScrubQuestionTypeId",
                "lastModifiedByUserId":"00000000-0000-0000-0000-000000000000",
                "lastModifiedDate":"scrubbed by ScrubLastModifiedDate",
                "questionType":{
                    "questionTypeId":"scrubbed by ScrubQuestionTypeId",
                    "description":"text",
                    "hasTextAnswer":true,
                    "hasBooleanAnswer":false
                },
                "answers":[
                    {
                        "answerId":"scrubbed by ScrubAnswerId",
                        "answerText":"00000000-0000-0000-0000-000000000000",
                        "answerBoolean":null,
                        "questionId":"scrubbed by ScrubQuestionId",
                        "lastModifiedDate":"scrubbed by ScrubLastModifiedDate",
                        "lastModifiedByUserId":"00000000-0000-0000-0000-000000000000",
                        "statusId":1,
                        "status":{
                            "statusId":1,
                            "name":"Draft"
                        },
                        "sourceURI":"consultations://./consultation/1/document/1/chapter/introduction"
                    }
                ],
                "commentId":"scrubbed by ScrubLocationId",
                "sourceURI":"consultations://./consultation/1/document/1/chapter/introduction",
                "htmlElementID":null,
                "rangeStart":null,
                "rangeStartOffset":null,
                "rangeEnd":null,
                "rangeEndOffset":null,
                "quote":null,
                "commentOn":"Chapter",
                "commentOnEnum":2,
                "documentId":1,
                "show":true,
                "order":"0",
                "section":null
            },
            {
                "questionId":"scrubbed by ScrubQuestionId",
                "questionText":"00000000-0000-0000-0000-000000000000",
                "questionTypeId":"scrubbed by ScrubQuestionTypeId",
                "lastModifiedByUserId":"00000000-0000-0000-0000-000000000000",
                "lastModifiedDate":"scrubbed by ScrubLastModifiedDate",
                "questionType":{
                    "questionTypeId":"scrubbed by ScrubQuestionTypeId",
                    "description":"text",
                    "hasTextAnswer":true,
                    "hasBooleanAnswer":false
                },
                "answers":[
                    
                ],
                "commentId":"scrubbed by ScrubLocationId",
                "sourceURI":"consultations://./consultation/1/document/1/chapter/introduction",
                "htmlElementID":null,
                "rangeStart":null,
                "rangeStartOffset":null,
                "rangeEnd":null,
                "rangeEndOffset":null,
                "quote":null,
                "commentOn":"Chapter",
                "commentOnEnum":2,
                "documentId":1,
                "show":true,
                "order":"0",
                "section":null
            }
        ],
        "isAuthorised":true,
        "signInURL":"/signin?returnURL=/",
        "consultationState":{
            "startDate":"2018-04-01T00:00:00",
            "endDate":"2028-05-31T17:00:00",
            "hasQuestions":false,
            "hasUserSuppliedAnswers":false,
            "hasUserSuppliedComments":false,
            "userHasSubmitted":false,
            "consultationSupportsQuestions":false,
            "consultationSupportsComments":false,
            "documentIdsWhichSupportQuestions":[
                2,
                2
            ],
            "documentIdsWhichSupportComments":[
                2,
                2
            ],
            "consultationIsOpen":true,
            "consultationHasNotStartedYet":false,
            "consultationHasEnded":false,
            "supportsSubmission":false,
            "supportsDownload":false,
            "shouldShowDrawer":true,
            "shouldShowCommentsTab":true,
            "shouldShowQuestionsTab":true
        }
    },
    "filters":[
        {
            "id":"Type",
            "title":"Response type",
            "options":[
                {
                    "id":"Questions",
                    "label":"Questions",
                    "isSelected":false,
                    "filteredResultCount":3,
                    "unfilteredResultCount":3
                },
                {
                    "id":"Comments",
                    "label":"Comments",
                    "isSelected":false,
                    "filteredResultCount":2,
                    "unfilteredResultCount":2
                }
            ]
        },
        {
            "id":"Document",
            "title":"Document",
            "options":[
                {
                    "id":"2",
                    "label":"Unstable angina and NSTEMI",
                    "isSelected":false,
                    "filteredResultCount":0,
                    "unfilteredResultCount":0
                },
                {
                    "id":"2",
                    "label":"Sinusitis (acute): antimicrobial prescribing",
                    "isSelected":false,
                    "filteredResultCount":0,
                    "unfilteredResultCount":0
                }
            ]
        }
    ],
    "organisationName":null,
    "type":null,
    "document":null,
    "sort":"DocumentAsc"
}